public/static/app/perfdiscovery/PerfDiscoveryDataService.js

```diff
@@ -69,32 +69,52 @@ mciModule.factory('PerfDiscoveryDataService', function(
     }
   }
 
-  function isPatchId(revision) {
-    return revision.length == EVG.PATCH_ID_LEN
+  function isShortItemId(id) {
+    return id.length == EVG.PATCH_ID_LEN
   }
 
-  function isGitHash(revision) {
-    return revision.length == EVG.GIT_HASH_LEN
+  function isLongVersionId(id) {
+    return id.indexOf('_') > 0 && id.length > EVG.PATCH_ID_LEN
   }
 
-  function getQueryBasedItem(items, query) {
-    var found = findVersionItem(items, query)
+  function isGitHash(id) {
+    return id.length == EVG.GIT_HASH_LEN
+  }
 
-    if (!found) {
-      var item = {
-        id: query,
+  function githashToVersionId(githash) {
+    return $window.project + '_' + githash
+  }
+
+  // For given `query` attempts to construct valid selctable
+  // comparison item. It could be version short/long id, githash
+  // Should not be used for tags (all tags are available on ui)
+  function getQueryBasedItem(query) {
+    if (query == undefined) return
```

Comment:
Should this check for both null and undefined?

Reply:
There shouldn't be any `null`s, but yes, it should check for both.